fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/item/ItemStorage.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.item;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.ChestBlock;
+import net.minecraft.block.InventoryProvider;
+import net.minecraft.block.entity.ChestBlockEntity;
+import net.minecraft.inventory.Inventory;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.Direction;
+
+import net.fabricmc.fabric.api.lookup.v1.block.BlockApiLookup;
+import net.fabricmc.fabric.api.transfer.v1.storage.Storage;
+
+/**
+ * Access to {@link Storage Storage&lt;ItemVariant&gt;} instances.
+ */
+public final class ItemStorage {
+	public static final BlockApiLookup<Storage<ItemVariant>, Direction> SIDED =
+			BlockApiLookup.get(new Identifier("fabric:sided_item_storage"), Storage.asClass(), Direction.class);
+
+	private ItemStorage() {
+	}
+
+	static {
+		// Load generic vanilla api fallback
+		ItemStorage.SIDED.registerFallback((world, pos, state, blockEntity, direction) -> {
```

Comment:
I would recommend just calling HopperBlockEntity#getInventory, that way if any mods have mixed into there or mix into there later or if mojang adds a new inventory getting system thing, compatibility will be maintained. Make BlockApiAccess or whatever it's called more powerful by letting it declare whether it needs the blockstate and block entity, this way we can half-avoid querying the block entity and blockstate, since HopperBlockEntity#getInventory does that internally.

Reply:
Block entities are cached by `BlockApiCache` already, making them very cheap to access in the providers.

If mods have a custom inventory system, they should just register a fallback provider in the lookup. If mojang changes something, the API will require updating but that doesn't sound like a big deal.

Reply:
Not suitable anymore anyway due to transactions requiring special handling. We can't support the full MC API.